python/singa/sonnx.py

```diff
@@ -0,0 +1,204 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+
+
+from __future__ import division
+import pickle
+from singa import tensor
+from singa.tensor import Tensor
+from singa import autograd
+from singa import optimizer
+import onnx
+from onnx import helper
+from onnx import AttributeProto, TensorProto, GraphProto
+from onnx import numpy_helper
+
+from collections import Counter, deque
+import math
+
+from .tensor import Tensor
+from . import layer
+from singa.proto import model_pb2
+from . import singa_wrap as singa
+#from .tensor import einsum
+from autograd import *
+
+def onnx_model_init(inputs,name):
+    '''
+    load onnx model graph and load weights
+    input:
+    input data and file name of onnx model
+
+    return:
+     a graph node dictionary
+     model: graph model
+    '''
+    f = open(name, 'rb')
+    model = pickle.load(f)
```

Comment:
ONNX models should not be loaded with pickle. Please use ONNX python API :` onnx_model = onnx.load('path/to/the/model.onnx')` instead. See [here](https://github.com/onnx/onnx/blob/master/docs/PythonAPIOverview.md#loading-an-onnx-model).

Reply:
yes, I changed to onnx.save and onnx.load